messageutil.go

```diff
@@ -0,0 +1,235 @@
+// the JSON sent over the network may not be populated fully.
+//
+//	message := anthropic.Message{}
+//	for stream.Next() {
+//		event := stream.Current()
+//		message.Accumulate(event)
+//	}
+func (acc *Message) Accumulate(event MessageStreamEventUnion) error {
+	if acc == nil {
+		return fmt.Errorf("accumulate: cannot accumlate into nil Message")
+	}
+
+	switch event := event.AsAny().(type) {
+	case MessageStartEvent:
+		*acc = event.Message
+	case MessageDeltaEvent:
+		acc.StopReason = event.Delta.StopReason
+		acc.StopSequence = event.Delta.StopSequence
+		acc.Usage.OutputTokens = event.Usage.OutputTokens
+	case ContentBlockStartEvent:
+		acc.Content = append(acc.Content, ContentBlockUnion{})
+		err := acc.Content[len(acc.Content)-1].UnmarshalJSON([]byte(event.ContentBlock.RawJSON()))
+		if err != nil {
+			return err
+		}
+	case ContentBlockDeltaEvent:
+		if len(acc.Content) == 0 {
+			return fmt.Errorf("received event of type %s but there was no content block", event.Type)
+		}
+		cb := &acc.Content[len(acc.Content)-1]
+		switch delta := event.Delta.AsAny().(type) {
+		case TextDelta:
+			cb.Text += delta.Text
+		case InputJSONDelta:
+			if len(delta.PartialJSON) != 0 {
+				if string(cb.Input) == "{}" {
+					cb.Input = []byte(delta.PartialJSON)
+				} else {
+					cb.Input = append(cb.Input, []byte(delta.PartialJSON)...)
+				}
+			}
+		case ThinkingDelta:
+			cb.Thinking += delta.Thinking
+		case SignatureDelta:
+			cb.Signature += delta.Signature
+		case CitationsDelta:
+			citation := TextCitationUnion{}
+			err := citation.UnmarshalJSON([]byte(delta.Citation.RawJSON()))
+			if err != nil {
+				return fmt.Errorf("could not unmarshal citation delta into citation type: %w", err)
+			}
+			cb.Citations = append(cb.Citations, citation)
+		}
+	case MessageStopEvent, ContentBlockStopEvent:
+		break
+	}
+
+	return nil
+}
+
+// ToParam converters
+
+func (r Message) ToParam() MessageParam {
+	var p MessageParam
+	p.Role = MessageParamRole(r.Role)
+	p.Content = make([]ContentBlockParamUnion, len(r.Content))
+	for i, c := range r.Content {
+		p.Content[i] = c.ToParam()
+	}
+	return p
+}
+
+func (r ContentBlockUnion) ToParam() ContentBlockParamUnion {
+	switch variant := r.AsAny().(type) {
+	case TextBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfText: &p}
+	case ToolUseBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfToolUse: &p}
+	case ThinkingBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfThinking: &p}
+	case RedactedThinkingBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfRedactedThinking: &p}
+	case ServerToolUseBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfServerToolUse: &p}
+	case WebSearchToolResultBlock:
+		p := variant.ToParam()
+		return ContentBlockParamUnion{OfWebSearchToolResult: &p}
+	default:
+		panic(fmt.Sprintf("unexpected anthropic.anyContentBlock: %#v", variant))
```

Comment:
Can we not `panic` and log something here instead and then return like we used to? 
```suggestion
	return ContentBlockParamUnion{}
```

There's some risk we forget a new content block type here. That or add a unit test to catch some cases.